apps/aecore/lib/aecore/structures/signed_tx.ex

```diff
@@ -30,12 +32,13 @@ defmodule Aecore.Structures.SignedTx do
     key == nil && signature == nil
   end
 
-  @spec validate(SignedTx.t()) :: boolean()
-  def validate(%{data: data, signature: signature}) do
-    case data do
-      %SpendTx{} ->
-        SpendTx.validate(data) &&
-          Signing.verify(Serialization.pack_binary(data), signature, data.from_acc)
+  @spec is_valid?(SignedTx.t()) :: boolean()
+  def is_valid?(%SignedTx{data: data} = tx) do
+    if Signing.verify(Serialization.pack_binary(data), tx.signature, data.from_acc) do
+      case DataTx.is_valid(data) do
```

Comment:
Why don't the is_valid functions for each type return a boolean so that there is no need for a case, the reason message isn't used anyway. If we want to tell the user the input is invalid, we can Log.error(reason) and return false after that.